cadence/gateway/cadence/interface.py

```diff
@@ -0,0 +1,11 @@
+from abc import abstractmethod
+from typing import Tuple
+
+from cadence.cadence_types import ListDomainsRequest, ListDomainsResponse
+
+
+class CadenceGatewayInterface:
```

Comment:
When it's ready, maybe switch to `WorkflowService` to be backward compatible with the existing one. 
And then the implementation classes can be named `TChannelWorkflowService` and `GrpcWorkflowService`. 
This is fitting our naming pattern in other SDK like Java. 

I feel like `Gateway` is more used as a server concept, this is just a client connecting to server.